.github/workflows/docker.yml

```
@@ -33,3 +33,16 @@ jobs:
           build-args: |
             GIT_REVISION=${{ steps.git.outputs.GIT_REVISION }}
             RUST_FEATURES=${{ matrix.rust-features }}
+      - name: Compose
+        id: compose
+        run: docker compose up --wait --wait-timeout 120
+        # continue on error so we can inspect containers in the next step
+        continue-on-error: true
+      - name: Inspect containers
+        if: steps.compose.outcome != 'success'
+        run: |
+          docker compose ps
+          for NAME in `docker compose ps --format json | jq -r '.Name'`; do
+            docker inspect $NAME
+          done
+          exit 1
```

We could simplify this by replacing the default `if: success()` condition on the last step, and getting rid of `continue-on-error: true`. See https://docs.github.com/en/actions/learn-github-actions/expressions#example-of-failure-with-conditions.
```suggestion
      - name: Inspect containers
        if: ${{ failure() && steps.compose.outcome != 'success' }}
        run: |
          docker compose ps
          for NAME in `docker compose ps --format json | jq -r '.Name'`; do
            docker inspect $NAME
          done
```

---

Oh, cool, I wasn't aware of this advance in the state of the art of encoding program logic as YAML